integration_tests/sdk/data_integration_tests/s3_test.py

```diff
@@ -569,3 +569,9 @@ def test_s3_save_failure(client, data_integration):
         InvalidUserArgumentException, match="Unsupported S3 file format `wrong format`."
     ):
         save(data_integration, hotel_reviews, generate_table_name(), format="wrong format")
+
+
+def test_s3_root_folder_present(client, data_integration):
```

Comment:
Actually, because root_dir only matters as artifact storage, it may be pretty hard to test as a specific case... Maybe you can just do a manual `test-config.yml` test run with a custom `root_dir` and verify that it passes?

Reply:
I have backed out the test-integration changes, and will create a separate issue for that.